### ISSUE TITLE
amp-list changes to input fields (i.e. autocomplete) aren't resolvable in amp-script

### ISSUE THREAD

**Issue by philkrie** [ampproject/worker-dom#653]
When amp-list is used to modify input fields (i.e. autocomplete), amp-script is unable to retrieve the modified input, and instead retrieves what the user typed.

You can see an example of the autocomplete in question [here](https://tryamp.msiteproject.com/bookit/index_test.html). If you type "PDX" into the To or From field, it autocompletes to "Portland, OR, OR, US". 

In the dev tools console, `document.getElementByID("to-input").value` resolves to `"Portland, OR, OR, US"`. In amp-script, it resolves to `"PDX"`.

(As noted in #628, amp-script has a limitation where you need to bind an event listener to the input field in order for it to access the value. This problem occurs with event listeners assigned to the autocompleted fields)

**Comment by philkrie** [comment]
I've noticed this behaviour recently with other AMP components as well, such as amp-date-picker. If I assign amp-date-picker to output to a input[type="date"] field that is not hidden, amp-script is unable to retrieve the value. If the user clicks on the input field and inputs a date, then amp-script will get the value.

**Comment by kristoferbaxter** [comment]
Leaving some raw notes here:

We might be able to resolve this by moving from an `oninput` property attached to anything that supports `change` events to using a mutation observer for the `value` attribute... which gets changed by `AMP.bind`.

```javascript
const observer = new MutationObserver(e => console.log(e.map(i => i.target.getAttribute('value'))));
observer.observe(el, {attributes: true});
```

We'll need to track the objects that have a `MutationObserver` attached to them, and remove the `MutationObserver`s when the `HTMLElement` is removed from the DOM.

```javascript
const observer = new MutationObserver((mutations, observer) => {
  if(mutations.type === 'childList' && //removed from DOM) {
    observer.disconnect();
  }
  console.log(mutations.map(i => i.target.getAttribute('value')));
});
observer.observe(el, {attributes: true, childList: true});

**Comment by kristoferbaxter** [comment]
Early version here: https://github.com/ampproject/worker-dom/pull/665

**Comment by kristoferbaxter** [comment]
Resolved with `0.21.0`.
